after gcc linking has changed, all the libraries must be explicitely specified to for linking. 
This patch avoids this linking error:

| make[2]: Entering directory `/disk0/pokybuild/build1/tmp/work/i586-poky-linux/matchbox-desktop-2.0+svnr2096-r0/matchbox-desktop-2/src'^M
| ccache i586-poky-linux-gcc -march=i586 --sysroot=/disk0/pokybuild/build1/tmp/sysroots/i586-poky-linux -Wall -fexpensive-optimizations -fomit-frame-pointer -frename-registers -O2 -ggdb -feliminate-unused-debug-types  -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -o matchbox-desktop main.o taku-category-bar.o desktop.o ../libtaku/libtaku.a -pthread -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lgdk_pixbuf-2.0 -lpangocairo-1.0 -lcairo -lpango-1.0 -lfreetype -lfontconfig -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0   -ldbus-1 -lpthread -lrt   -lstartup-notification-1   ../libtaku/libinotify.a^M
| /disk0/pokybuild/build1/tmp/sysroots/x86_64-linux/usr/libexec/i586-poky-linux/gcc/i586-poky-linux/4.5.1/ld: F^A: invalid DSO for symbol `XFree' definition^M
| /disk0/pokybuild/build1/tmp/sysroots/i586-poky-linux/usr/lib/libX11.so.6: could not read symbols: Bad value^M
| collect2: ld returned 1 exit status^M
| make[2]: *** [matchbox-desktop] Error 1

Nitin A Kamble <nitin.a.kamble@intel.com>
Date: 2011/01/11

Upstream-Status: Accepted

Index: matchbox-desktop-2/configure.ac
===================================================================
--- matchbox-desktop-2.orig/configure.ac
+++ matchbox-desktop-2/configure.ac
@@ -14,7 +14,7 @@ if test x$inotify_support = xyes; then
         AC_DEFINE(WITH_INOTIFY, [1], [If inotify is enabled])
 fi
 
-PKG_CHECK_MODULES(GTK, gtk+-2.0)
+PKG_CHECK_MODULES(GTK, [gtk+-2.0 x11])
 
 AC_ARG_ENABLE(startup_notification,
         AC_HELP_STRING([--disable-startup-notification], [disable startup notification support]),
